sds/src/scanner/mod.rs

```diff
@@ -1036,17 +1039,21 @@ mod test {
 
         let mut content = SimpleEvent::Map(BTreeMap::from([
             (
-                "message".to_string(),
+                "a-match".to_string(),
+                SimpleEvent::String("bcdef".to_string()),
+            ),
+            (
+                "z-match".to_string(),
                 SimpleEvent::String("abcdef".to_string()),
             ),
             ("test".to_string(), SimpleEvent::String("bcdef".to_string())),
         ]));
 
         let matches = scanner.scan(&mut content);
 
-        // The match from the "test" field (which is excluded) is the same as the match from "message", so it is
-        // treated as a false positive.
-        assert_eq!(matches.len(), 0);
+        // Due to the ordering of the scan (alphabetical in this case), the match from "a-match" is not
+        // excluded yet, but "z-match" is, so only 1 match is found.
+        assert_eq!(matches.len(), 1);
```

Comment:
Minor: let's validate that the path of the match is `a-match` so it's more explicit that `z-match` is the one excluded.

Reply:
Fixed in a9d44ba1848f996b99c3d8a4bb83e59d8af51ba6